packages/slice-machine/components/Simulator/index.tsx

```diff
@@ -16,12 +13,24 @@ import {
   selectSimulatorUrl,
 } from "@src/modules/environment";
 import { SliceMachineStoreType } from "@src/redux/type";
+import { selectCurrentSlice } from "@src/modules/selectedSlice/selectors";
+import Router from "next/router";
 
 export type SliceView = SliceViewItem[];
 export type SliceViewItem = Readonly<{ sliceID: string; variationID: string }>;
 
 export default function Simulator() {
-  const { Model, variation } = useContext(SliceContext);
+  const { Model } = useSelector((store: SliceMachineStoreType) => ({
+    Model: selectCurrentSlice(
+      store,
+      Router.router?.query.lib as string,
```

Comment:
yes but only info about the `location` which would just be a string, this means it becomes hard to query for the lib or sliceName within the url.

Also the location is not always kept in redux since the redirection when clicking on a slice or custom-type is not done through `connected-next-router`. But think that's outside of this ticket scope since this was to refactor library and slice context

Reply:
good to know :)